tuplex/core/src/physical/TransformStage.cc

```diff
@@ -746,12 +746,15 @@ namespace tuplex {
         JobMetrics dummy_metrics;
         JobMetrics& metrics = PhysicalStage::plan() ? PhysicalStage::plan()->getContext().metrics() : dummy_metrics;
 
+        auto unoptimizedIR = code();
```

Comment:
I presume this function can be quite expensive to call, maybe only execute when history server is enabled? (same for the optimized one)

Reply:
Done. Refactored to be in a conditional checking for WebUI.